bokehjs/src/lib/models/widgets/file_input.ts

```diff
@@ -16,43 +19,68 @@ export class FileInputView extends WidgetView {
     if (this.dialogEl == null) {
       this.dialogEl = document.createElement('input')
       this.dialogEl.type = "file"
-      this.dialogEl.multiple = false
-      this.dialogEl.onchange = (e) => this.load_file(e)
+      this.dialogEl.multiple = false  
+      this.dialogEl.onchange = async (e) => await this.load_files(e)
       this.el.appendChild(this.dialogEl)
     }
     if (this.model.accept != null && this.model.accept != '')
       this.dialogEl.accept = this.model.accept
+    if (this.model.multiple == 'multiple' || this.model.multiple == 'True'){
+      this.dialogEl.multiple = true
+    }
     this.dialogEl.style.width = `{this.model.width}px`
     this.dialogEl.disabled = this.model.disabled
   }
 
-  load_file(e: any): void {
-    const reader = new FileReader()
-    this.model.filename = e.target.files[0].name
-    reader.onload = (e) => this.file(e)
-    reader.readAsDataURL(e.target.files[0])
-  }
-
-  file(e: any): void {
-    const file = e.target.result
-    const file_arr = file.split(",")
-
-    const content = file_arr[1]
-    const header = file_arr[0].split(":")[1].split(";")[0]
-
-    this.model.value = content
-    this.model.mime_type = header
 
+  // multiple = True
+  async load_files(e: any): Promise<void> {
+    const files = e.target.files
+    var i : number
+    var value : string[] = []
+    var filename : string[] = []
+    var mime_type : string[] = []
+    
+    for (i=0; i< files.length ; i++){
+      filename.push(files[i].name)
+      let content = await this.readfile(files[i])
+      const file_arr = content.split(",")
+      value.push(file_arr[1])
+      mime_type.push(file_arr[0].split(":")[1].split(";")[0])
+      //console.log(content);
+    }
+    if (this.model.multiple == 'multiple' || this.model.multiple == 'True'){
+      this.model.filename = filename
+      this.model.mime_type= mime_type
+      this.model.value = value
+    }
+    else {
+      this.model.filename = filename[0]
+      this.model.mime_type= mime_type[0]
+      this.model.value = value[0]
+    }
+    //console.log('load_files:')
+    //console.log(this.model.filename)
+    //console.log(this.model.value)
+  }
+  
+  readfile(file: any): Promise<string> { //<string | ArrayBuffer | null>
+    return new Promise<any>((resolve) => {
+      const reader = new FileReader()
+      reader.onload = (e) => resolve(e.target ? e.target.result : "")
+      reader.readAsDataURL(file)
+    })
```

Comment:
All this and other code needs to be cleaned up, before we can get to any detailed review. You will need to go through some other bokehjs' `*.ts` files, familiarize yourself with how we structure out source code and then apply this knowledge here.

Reply:
Here, I expect larger complications, because I'm completely new to TS/JS (I tried to avoid it by using bokeh, hah). I have little hope that I am actually capable to even detect structural or stylistic differences. Not to mention that the your meaning of "structural" is not completely clear to me, sorry.
I can try to comply.

Reply:
Not sure what to do, continue discussion in PR #9732 ?!